builder/docker/communicator.go

```diff
@@ -41,7 +68,13 @@ func (c *Communicator) Start(remote *packer.RemoteCmd) error {
 	// This file will store the exit code of the command once it is complete.
 	exitCodePath := outputFile.Name() + "-exit"
 
-	cmd := exec.Command("docker", "attach", c.ContainerId)
+	var cmd *exec.Cmd
+	if useDockerExec {
+		cmd = exec.Command("docker", "exec", "-i", c.ContainerId, "/bin/sh")
```

Comment:
This worked for me, but then I ran into the `you must have a tty to run sudo` on CentOS. I recommend adding `-t` option as well.